[CCR] Add Remote Cluster form

cjcenizal opened the pull request:
Includes rudimentary client-side validation.

![image](https://user-images.githubusercontent.com/1238659/48382606-a5721480-e696-11e8-87cd-64cf966aa342.png)

![image](https://user-images.githubusercontent.com/1238659/48382610-a7d46e80-e696-11e8-8099-5de3c861c1ca.png)

elasticmachine commented:
Pinging @elastic/kibana-management

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/8672/)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/8676/)

cjcenizal commented on x-pack/plugins/remote_clusters/public/sections/remote_cluster_add/remote_cluster_form.js:
```diff
@@ -0,0 +1,349 @@
+      let wrappedField;
+
+      if (index === 0) {
+        wrappedField = field;
```
Good idea about EuiComboBox! I'll try that out.

cjcenizal commented:
@jen-huang I've implemented your EuiComboBox suggestion:

![image](https://user-images.githubusercontent.com/1238659/48514520-56052300-e813-11e8-8b84-39ecf4f1f9b6.png)

![image](https://user-images.githubusercontent.com/1238659/48514523-57cee680-e813-11e8-8254-7c82b1cddfb9.png)

I added some validation to restrict IP addresses to IPv4 format, but I'm not sure if that's a good idea. I'll merge this for now and follow up on these validation rules with the ES team.

jen-huang commented:
`EuiComboBox` looks good! Agreed, we'll have to adjust the validation. IPv6 might be allowed and possibly standard hostnames as well. Ports are definitely allowed (like `127.0.0.1:9400`).

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/94/)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/110/)
